rdflib/plugins/parsers/trig.py

```diff
@@ -69,16 +69,20 @@ def graph(self, argstr: str, i: int) -> int:
         raise Exception if it looks like a graph, but isn't.
         """
 
+        need_graphid = False
         # import pdb; pdb.set_trace()
         j = self.sparqlTok("GRAPH", argstr, i)  # optional GRAPH keyword
         if j >= 0:
             i = j
+            need_graphid = True
 
         r: MutableSequence[Any] = []
         j = self.labelOrSubject(argstr, i, r)
         if j >= 0:
             graph = r[0]
             i = j
+        elif need_graphid:
+            return -1
```

Comment:
```suggestion
            self.BadSyntax(argstr, i, "GRAPH keyword must be followed by graph name")
```

I think it may be best to also use `self.BadSyntax` here, as it makes the problem a bit clearer to users.